6/99_hw/db_explorer/db_explorer.go

```diff
@@ -2,3 +2,554 @@ package main
+			fmt.Println("\tCould not find requested element with ID=" + itemId)
+			var response = DataToExchange{
+				"error": "record not found",
+			}
+			w.WriteHeader(http.StatusNotFound)
+			w.Write(response.Bytes())
+			return
+		}
+
+		var response = DataToExchange{
+			"response": DataToExchange{
+				"record": records[0],
+			},
+		}
+
+		w.WriteHeader(http.StatusOK)
+		w.Write(response.Bytes())
+	}
+}
+
+func (h *Handler) processPutRequest(w http.ResponseWriter, r *http.Request) {
+	paths := strings.Split(strings.Trim(r.URL.Path, "/"), "/")
+	if len(paths) != 1 {
+		w.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+
+	var tableName = paths[0]
+	if !h.checkTableExistance(w, tableName) {
+		return
+	}
+
+	body, err := io.ReadAll(r.Body)
+	if err != nil {
+		w.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+	defer r.Body.Close()
+
+	req := DataToExchange{}
+	err = json.Unmarshal(body, &req)
+	if err != nil {
+		w.WriteHeader(http.StatusInternalServerError)
+		fmt.Println("\tUnable to unmarshal body:", err)
+		return
+	}
+
+	var info = h.Tables[tableName]
+	var columnsToInsert string
+	var substitutionMask string
+	var values = make([]interface{}, 0, len(info.Columns))
+	var firstCol bool = true
+	for _, colInfo := range info.Columns {
+		value, found := req[colInfo.Name]
+		if !found {
+			// The column has no value presented in
+			// the request so generate value when there is
+			// no default value provided in the DB.
+			if colInfo.DefaultValue != nil {
+				value = colInfo.DefaultValue
+			} else if !colInfo.MayBeNull {
+				value = defaultValueForType(colInfo.Type)
+			}
+		}
+
+		if !firstCol {
+			columnsToInsert += ", "
+			substitutionMask += ", "
+		}
+		firstCol = false
+
+		columnsToInsert += colInfo.Name
+		substitutionMask += "?"
+		values = append(values, value)
+	}
+
+	var insertStatement = fmt.Sprintf("INSERT INTO %s (%s) VALUES (%s)", tableName, columnsToInsert, substitutionMask)
+	result, err := h.Db.Exec(insertStatement, values...)
+	if err != nil {
+		w.WriteHeader(http.StatusInternalServerError)
+		fmt.Printf("%s", errors.Wrap(err, fmt.Sprintf("unable to insert into %s", tableName)))
+		return
+	}
+
+	lid, _ := result.LastInsertId()
+	fmt.Println("\tInserted item id:", lid)
+
+	var pkey = h.Tables[tableName].getPrimaryKeyColumn()
+	var response = DataToExchange{
+		"response": DataToExchange{
+			pkey: lid,
+		},
+	}
+
+	w.WriteHeader(http.StatusOK)
+	w.Write(response.Bytes())
+}
+
+func (h *Handler) processPostRequest(w http.ResponseWriter, r *http.Request) {
+	paths := strings.Split(strings.Trim(r.URL.Path, "/"), "/")
+	if len(paths) != 2 {
+		w.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+
+	var tableName = paths[0]
+	var itemId = paths[1]
+	if !h.checkTableExistance(w, tableName) {
+		return
+	}
+
+	body, err := io.ReadAll(r.Body)
+	if err != nil {
+		w.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+
+	defer r.Body.Close()
+
+	req := DataToExchange{}
```

Comment:
Вот тут как раз неинициализированная переменная, её через var бы
```
var req DataToExchange
```suggestion
	var req DataToExchange
```